File: Lecture 3/practice/PrimeList.py
#2. Liệt kê các số nguyên tố nhỏ hơn n
import math


def isPrime(num):
    if num < 2:
        return False
    for i in range(2, int(math.sqrt(num) + 1)):
        if num % i == 0:
            return False
    return True


def listPrime(num):
    if num <= 2:
        return []
    else:
        prime = listPrime(num - 1)
        if isPrime(num - 1):
            prime.append(num - 1)
        return prime


num = int(input("Upperbound:"))
print(listPrime(num))

File: Lecture 3/practice/PrimeList.java

public class PrimeList {
    
    public static boolean isPrime(int num){
        if (num < 2) 
            return false;
        for(int i = 2; i * i <= num; i++){
            if(num % i == 0){
                return false;
            }
        }
        return true;
    }

    public static void listPrime(int num, int prime){
        if(prime < num){
            if(isPrime(prime)){
                System.out.print(prime + " ");
            }
            listPrime(num, prime + 1);
        }
    }

    public static void main(String[] args) {
        int num = 20;
        System.out.println("Prime numbers less than " + num + " are: ");
        listPrime(num, 2);
    }
}
